contrib/pyln-client/pyln/client/gossmap.py

```diff
@@ -273,12 +273,97 @@ def get_halfchannel(self,
         channel = self.get_channel(short_channel_id)
         return channel.half_channels[direction]
 
+    def get_halfchannels(self,
```

Comment:
Sounds more like `neighborhood(source: ..., n: int)`. Also this has way too much magic, have it return the channels, and then have the caller map to directions, and then have them filter it to make it directed paths.

Reply:
The naming was pretty dumb for both methods. For now I renamed it to `get_neightbors_hc` (hc for half-channel variant). Not sure if we really need this method, the other one (now `get_neighbors`) is more useful I guess. Let me explain:

It returns a set of half-channels connected from or towards (using direction) your neightborhood (at a given distance). So an outer shell of half-channels of that neighborhood. This can be used to do some math on how good the connectivity is at a certain distance (e.g. in terms of median fees, channel size, etc..).